src/python/pants/core/target_types.py

```diff
@@ -858,9 +861,19 @@ async def package_archive_target(field_set: ArchiveFieldSet) -> BuiltPackage:
 # -----------------------------------------------------------------------------------------------
 
 
-class LockfileSourceField(SingleSourceField):
+class LockfileSourceField(OptionalSingleSourceField):
+    """Source field for synthesized `_lockfile` targets.
+
+    It is special in that it always ignores any missing files, regardless of the global
+    `--unmatched-build-file-globs` option.
+    """
+
     uses_source_roots = False
     required = True
+    value: str
+
+    def path_globs(self, unmatched_build_file_globs: UnmatchedBuildFileGlobs) -> PathGlobs:  # type: ignore[misc]
+        return super().path_globs(UnmatchedBuildFileGlobs.ignore())
```

Comment:
Maybe a comment explaining why?

Reply:
Not worth a CI round though. Feel free to merge as-is.

Reply:
Ah, yea I figured the field class docstring would be that, but see it doesn't say why only what. But yea, it'll be easy enough to pull up this PR with a `git blame`.